### PR TITLE
Slider not working inside a flex container

### LINKED ISSUE #1750: Slider images not rendered if added in Neve's header
### Description

A customer reported that when he adds a Slider block in Customizer -> Header -> Header Builder -> Widget Area 1 (for both desktop and mobile view), the images are displayed correctly on desktop, but only a blank area appears on mobile. When I test this on my instance, the images don't appear on desktop, but they do on mobile. There is no error in the Console. He also encounters the issue when he uses a slider from the Smart Slider 3 plugin.

### Step-by-step reproduction instructions

1. Install Neve, Neve Pro and Otter
2. Go to Customizer -> Header -> Header Builder
3. Add a Widget Area 1 component
4. Add a Slider block in that component
5. Repeat steps 3-4 for mobile view as well
6. Check the slider images on desktop and mobile

### Screenshots, screen recording, code snippet or Help Scout ticket

Desktop view on the client's site:

desktop2

Mobile view on the client's site:

mobile2

Desktop view on my instance:

desktop

Mobile view on my instance:

mobile

Reported [here](https://secure.helpscout.net/conversation/2289018013/375912/) .

### Environment info

_No response_

### Is the issue you are reporting a regression

No

### PR DESCRIPTION
**Pull request by cristian-ungureanu**
Closes #1750, Codeinwp/neve#4092.

### Summary
- Due to a [Glide bug](https://github.com/glidejs/glide/issues/534), the slider is stretching far more than the container if the container has a flexible width.
- It seems that making the container "display:grid" fixes the issue
- The fix was already in Otter but it only worked if you would have added the Slider [inside a column](https://github.com/Codeinwp/otter-blocks/blob/master/src/blocks/blocks/section/style.scss#L77-L79)

### Test instructions
- Install Neve, Neve Pro and Otter
- Add a widget section inside Neve's header and inside that widget section add a Slider
- Save the customizer and check it on frontend
- Please check https://github.com/Codeinwp/otter-blocks/issues/1750 and also https://github.com/Codeinwp/neve/issues/4092 and try to see if the problem is solved

Please note that the issue I've mentioned [here](https://github.com/Codeinwp/neve/issues/4092#issuecomment-1727748053) is still there but I think we can consider it an edge case. Here's how it manifests: https://vertis.d.pr/v/Nm55m6

---- 

### Checklist before the final review

- [ ] Included E2E or unit tests for the changes in this PR.
- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.
- [ ] Copy/Paste is working if the attributes are modified.
- [ ] PR is following [the best practices]()

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	"packages": {
 		"": {
 			"name": "otter-blocks",
-			"version": "2.3.4",
+			"version": "2.4.0",
```
[the remark this answers is not in the document]
Comment (cristian-ungureanu) [reply package-lock.json:9]: Development. Not sure what happened here, it's 2.3.4 on both master and development. I just ran `npm install`. Probably I should've added --frozen-lockfile flag. I'll revert the change, sorry for this.

[the remark this answers is not in the document]
Comment (Soare-Robert-Daniel) [reply package-lock.json]: We avoid `install` since it is trying to update the packages. `npm ci` is the way to install the deps without any modification.

**Review by Soare-Robert-Daniel — APPROVED** [review]
Your solution is 🔥 . Also, it is not breaking for me in Chrome, Vivaldi, Firefox, and Safari 💯 
https://github.com/Codeinwp/otter-blocks/assets/17597852/b9056dd2-9691-4e1d-b004-ad5240c8fef9

**Comment by rodica-andronache** [comment]
hey @cristian-ungureanu I'm not sure what I'm missing, but the slider is not working for me. I've tried both on instawp and local. https://vertis.d.pr/v/vL02Tl
You can check it here:
```
https://yellowed-yacare-oegik.vertisite.cloud/
user: admin
pass: yyF@^@tgUQuReny#4MmjU*cHmim
```

**Comment by cristian-ungureanu** [comment]
Hey @rodica-andronache, thanks for testing. I opened another ticket for it here: https://github.com/Codeinwp/otter-blocks/issues/1907, it seems the files are not loaded if you add a Slider block in the header like this.

When I tested it, I had another slider block inside my page, that's why I didn't see this issue. Hardeep or Robert should be able to handle it. Until then, to test this particular issue, you can add another slider inside the page content.

**Comment by rodica-andronache** [comment]
@cristian-ungureanu ok, tested again with another slider in the content of the page, and I can confirm that **in that case**, the slider from the header is loading as well.
But there is another issue. The arrows of the slider from the header area are not working, on **mobile** https://vertis.d.pr/v/2djypx

**Comment by Soare-Robert-Daniel** [comment]
Now, the slider should work in Neve Header on Mobile.

The solution was not noticeable at first.

When you use the header builder for desktop and mobile, and add the same widget then the slider will appear in two places: the section for desktop and the section for mobile.

Because we were using a dump selector like `#${ slider.id }`, we were only getting the first apparition of the slider, which in the context of the builder was the desktop one. This made that the mobile slider to be left out.

We should avoid as much as possible to make selectors like this because it took a noticeable time to debug.

**Review by HardeepAsrani — APPROVED** [review]
